File: lib_common/src/main/java/com/style/entity/UserInfo.java
package com.style.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(tableName = "user_info_table", foreignKeys = @ForeignKey(entity = UserBean.class, parentColumns = "id", childColumns = "user_id", onDelete = ForeignKey.CASCADE))
public class UserInfo implements Serializable {

    @PrimaryKey
    @ColumnInfo(name = "user_id")
    public String userId;
    @ColumnInfo(name = "tel_phone")
    public String telPhone;
    @ColumnInfo(name = "password")
    public String password;
    @ColumnInfo(name = "username")
    public String userName;
    @ColumnInfo(name = "sex")
    public String sex;
    @Ignore
    public String signKey;

    public UserInfo() {
    }

    public UserInfo(String sex) {
        this.sex = sex;
    }

    public UserInfo(String telPhone, String password) {
        this.telPhone = telPhone;
        this.password = password;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getTelPhone() {
        return telPhone;
    }

    public void setTelPhone(String telPhone) {
        this.telPhone = telPhone;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getSex() {
        return sex;
    }

    public void setSex(String sex) {
        this.sex = sex;
    }

    public String getSignKey() {
        return signKey;
    }

    public void setSignKey(String signKey) {
        this.signKey = signKey;
    }

    @Override
    public String toString() {
        return "UserInfo{" +
                "userId='" + userId + '\'' +
                ", telPhone='" + telPhone + '\'' +
                ", password='" + password + '\'' +
                ", userName='" + userName + '\'' +
                ", sex='" + sex + '\'' +
                ", signKey='" + signKey + '\'' +
                '}';
    }
}

File: lib_common/src/main/java/com/style/entity/UserInfo.kt
package com.style.entity

import android.arch.persistence.room.*
import android.arch.persistence.room.ForeignKey.CASCADE
import java.io.Serializable

@Entity(tableName = "user_info_table",
        foreignKeys = [ForeignKey(entity = UserBean::class, parentColumns = arrayOf("id"), childColumns = arrayOf("user_id"), onDelete = CASCADE)])
open class UserInfo(var sex: String?) : Serializable {

    @PrimaryKey
    @ColumnInfo(name = "user_id")
    var userId: String? = null
    @ColumnInfo(name = "tel_phone")
    var telPhone: String? = null
    @ColumnInfo(name = "password")
    var password: String? = null
    @ColumnInfo(name = "username")
    var userName: String? = null
    @Ignore
    var signKey: String? = null

    constructor() : this(null)

    constructor(userId: String, password: String) : this() {
        this.userId = userId
        this.password = password
    }

}

data class SubUserInfo(var weight: String) : UserInfo("") {

}